### ISSUE TITLE
Music from youtube doesn't show up in search results

### ISSUE THREAD

**Issue by Rijaja** [futo-org/grayjay-android#1105]
I'm referring to music that is uploaded to youtube as a track rather than as a video. For example:
- [JY-HSMWkRYM](https://www.youtube.com/watch?v=JY-HSMWkRYM) is a track. You can see the description mentions "Auto-generated by YouTube.". It also has a square aspect ratio, because it's just a picture, not a video.
- [cTn288M5Mak](https://www.youtube.com/watch?v=cTn288M5Mak), on the other hand, is a user-uploaded video.

You can also see the difference when doing [a search in youtube music](https://music.youtube.com/search?q=darkshines+muse): songs and videos are two separate categories

I have successfully played many of those tracks through a playlist that was already on my youtube account. But when searching for them in the grayjay app, they never show up, even when they're the first result of the youtube search.

I'm guessing it's what #342 was looking for when asking for a "plugin for youtube music".

This is not new, I've been using grayjay for a while now, and I don't remember this working in the past.

Grayjay 242
Android 10

**Comment by codebymatter** [comment]
Artist - Topic specific channels are purposely hidden on YouTube search but aometimes appear. They appear more often on a seaech engine via browser, which is what I've been doing for 7-8 months in order to add them to a custom playlist.

**Comment by Rijaja** [comment]
Most of the videos I've searched for were not on "topic" channels. The example I provided comes from the official Muse channel.

Do you have any clue as to why they would hide them on purpose?

**Comment by codebymatter** [comment]
> Most of the videos I've searched for were not on "topic" channels. The example I provided comes from the official Muse channel.
> 
> Do you have any clue as to why they would hide them on purpose?

Yes as is stated in your original post. I should've clarified that I meant to add as extra, apart from your post ("on top of"). As to why Artist - Topic videos are hidden from YouTube search is beyond me, no clue. It's possible those "videos" or songs are specifically meant for YT Music app. If you visit the "Artist - Topic" labeled page on YT you will not see anything on there. Note that many of these songs with just album art covers also contained the text. My only theory will be that G has a say in how specific content are easily viewable. But even then, it's quite baseless as they can be found through a Search Engine outside of Grayjay and will sometimes appear as a search result inside of YT URL/mobile/app.

**Comment by Rijaja** [comment]
I think I found the code that makes these videos not show up in [line 3522](https://github.com/futo-org/grayjay-plugin-youtube/blob/5032e4e10a353324aeac582d58fef35c1743e967/YoutubeScript.js#L3522):
```js
if(!isLive && !videoRenderer.publishedTimeText?.simpleText)
		return  null; //Not a normal video
```
When using the youtube search engine, you can clearly see that there is no date on the video I'm looking for:
![image](https://github.com/futo-org/grayjay-android/assets/33258902/f3f93758-8735-4da5-8839-c061c52feed8)

I'll try modifying the plugin and see if that fixes my issue.
